java-frontend/src/main/java/org/sonar/java/se/FlowComputation.java

```diff
@@ -162,14 +152,15 @@ public NodeSymbol(@Nullable ExplodedGraph.Node node, @Nullable Symbol trackSymbo
   private Stream<JavaFileScannerContext.Location> methodInvocationFlow(Constraint learnedConstraint, ExplodedGraph.Node currentNode, ExplodedGraph.Node parent) {
     MethodInvocationTree mit = (MethodInvocationTree) parent.programPoint.syntaxTree();
     Stream.Builder<JavaFileScannerContext.Location> flowBuilder = Stream.builder();
-    if (currentNode.programState.peekValue() == symbolicValue) {
+    SymbolicValue returnSV = currentNode.programState.peekValue();
+    if (symbolicValues.contains(returnSV)) {
       flowBuilder.add(location(parent, String.format("'%s()' returns %s.", mit.symbol().name(), learnedConstraint.valueAsString())));
     }
-    SymbolicValue methodIdentifier = parent.programState.peekValue(mit.arguments().size());
-    if (methodIdentifier == symbolicValue) {
+    SymbolicValue invocationTarget = parent.programState.peekValue(mit.arguments().size());
+    if (symbolicValues.contains(invocationTarget)) {
       flowBuilder.add(location(parent, "..."));
     }
-    int argIdx = correspondingArgumentIndex(symbolicValue, parent);
+    int argIdx = correspondingArgumentIndex(symbolicValues, parent);
```

Comment:
I don't see how this could work nicely in a general way. I am pretty sure we can construct cases where the FIXME in correspondingArgumentIndex will be match. And that means that instead of changing its signature to admit a set we should loop over the set and call it (and handle the result) for each SV we are interested in.